### PR TITLE
[2128] Make copyright for generated classes consistent

### LINKED ISSUE #2128: Make copyright for generated classes consistent
Today, only copyright from EMF generated classes are not with the format :

```
/*******************************************************************************
 * Copyright (c) ${year} Obeo.
 * This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v2.0
 * which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *     Obeo - initial API and implementation
 *******************************************************************************/
``` 
but 

```
/**
 * Copyright (c) ${year} Obeo.
 * This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v2.0
 * which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *      Obeo - initial API and implementation
 */
```

Change them to a single format.

### PR DESCRIPTION
**Pull request by frouene**
Bug: https://github.com/eclipse-sirius/sirius-components/issues/2128

## General purpose
What is the main goal of this pull request?
- [ ] Bug fixes
- [ ] New features
- [ ] Documentation
- [x] Cleanup
- [ ] Tests
- [ ] Build / releng

## Project management
- [ ] Has the pull request been added to the relevant project and milestone? (Only if you know that your work is part of a specific iteration such as the current one)
- [ ] Have the `priority:` and `pr:` labels been added to the pull request? (In case of doubt, start with the labels `priority: low` and `pr: to review later`)
- [ ] Have the relevant issues been added to the pull request?
- [ ] Have the relevant labels been added to the issues? (`area:`, `difficulty:`, `type:`)
- [ ] Have the relevant issues been added to the same project and milestone as the pull request?
- [ ] Has the `CHANGELOG.adoc` been updated to reference the relevant issues?
- [ ] Have the relevant API breaks been described in the `CHANGELOG.adoc`? (Including changes in the GraphQL API)
- [ ] In case of a change with a visual impact, are there any screenshots in the `CHANGELOG.adoc`? For example in `doc/screenshots/2022.5.0-my-new-feature.png`

## Architectural decision records (ADR)
- [ ] Does the title of the commit contributing the ADR start with `[doc]`?
- [ ] Are the ADRs mentioned in the relevant section of the  `CHANGELOG.adoc`?

## Dependencies
- [ ] Are the new / upgraded dependencies mentioned in the relevant section of the `CHANGELOG.adoc`?
- [ ] Are the new dependencies justified in the `CHANGELOG.adoc`?

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose
- [ ] Is the code properly tested? (Plain old JavaScript tests for business code and tests based on React Testing Library for the components)

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- [ ] Variables have a proper type
- [ ] Functions’ arguments have a proper type
- [ ] Functions’ return type are specified
- Hooks are properly typed:
	- [ ] `useMutation<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useQuery<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useSubscription<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useMachine<CONTEXT_TYPE, EVENTS_TYPE>(…)`
	- [ ] `useState<STATE_TYPE>(…)`
- [ ] All components have a proper typing for their props
- [ ] No useless optional chaining with `?.` (if the GraphQL API specifies that a field cannot be `null`, do not treat it has potentially `null` for example)
- [ ] Nullable values have a proper type (for example `let diagram: Diagram | null = null;`)

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose
- [ ] Are all the event handlers tested?
- [ ] Are the event processor tested?
- [ ] Is the business code (services) tested?
- [ ] Are diagram layout changes tested?

### Architecture
- [ ] Are data structure classes properly separated from behavioral classes?
- [ ] Are all the relevant fields final?
- [ ] Is any data structure mutable? If so, please write a comment indicating why
- [ ] Are behavioral classes either stateless or side effect free?

### How to test this PR?
_Please describe here the various use cases to test this pull request_

- [ ] Has the Kiwi TCMS test suite been updated with tests for this contribution?

### CODE REVIEW

**Review by sbegaudeau — CHANGES_REQUESTED** [review]
Could you please update `GeneralPurposeTests` to start considering the projects with generated code like `sirius-components-domain`, `sirius-components-view`, `sirius-web-customwidgets` and `sirius-components-widget-reference-view`.

If we want consistent copyright headers, then I would prefer to let the code used to ensure the consistency of the copyright headers do the verification to make sure that the changes done here are correct and that they won't be ignored later.

**Comment by sbegaudeau** [comment]
Could you also take a second to link your issue to the pull request:

Capture d’écran 2023-07-03 à 10 30 35

On top of that, could you also reference this issue in the Github project to help us track the work done in the iteration, thanks.

File: packages/tests/backend/sirius-components-tests/src/test/java/org/eclipse/sirius/components/tests/GeneralPurposeTests.java [hunk 1]
```diff
@@ -171,8 +171,8 @@ private List<Path> findFilePaths(Path sourceFolderPath, String extension) {
     public void checkJavaCode() {
         File rootFolder = this.getRootFolder();
         Path backendFolderPath = Paths.get(rootFolder.getAbsolutePath(), BACKEND_FOLDER_PATH);
-        List<Path> javaFilePaths = this.findFilePaths(backendFolderPath, JAVA_FILE_EXTENSION);
-        for (Path javaFilePath : javaFilePaths) {
+        List<Path> javaFilePathsWithoutGenerated = this.findFilePaths(backendFolderPath, JAVA_FILE_EXTENSION, true);
```
Comment (sbegaudeau) [inline packages/tests/backend/sirius-components-tests/src/test/java/org/eclipse/sirius/components/tests/GeneralPurposeTests.java]: Don't scan and read every single line of every single Java files in the project twice. Put the filter to select the tests to perform `path.contains("/sirius-components-domain"))` inside of the for loop performing the various tests

**Review by sbegaudeau — CHANGES_REQUESTED** [review]

File: packages/tests/backend/sirius-components-tests/src/test/java/org/eclipse/sirius/components/tests/GeneralPurposeTests.java (hunk 1 above)
Comment (frouene) [reply packages/tests/backend/sirius-components-tests/src/test/java/org/eclipse/sirius/components/tests/GeneralPurposeTests.java]: Done

**Review by sbegaudeau — APPROVED** [review]
I'll merge it after the build

**Pull request merged**